### ISSUE TITLE
Timeshift performed a system restore without me doing it

### ISSUE THREAD

**Issue by RickCHodgin** [teejee2008/timeshift#748]
**Note:** This application is provided "as-is" without support or warranty. The purpose of this tracker is to keep track of issues. It is checked once a year when the app is being updated. If you face any issue you can report it here but please keep in mind that there is no dedicated support. The issue will be investigated and fixed during the next update which may occur many months after you report the issue.

**Note:** If you are a Linux Mint user and you need help with an issue, consider posting the issue on [Linux Mint forums](https://forums.linuxmint.com) where there are more people to help you out.

**Describe the bug**
1. I shut down my computer one day.
2. I booted my computer the next day and it took a long time to boot.
3. When it had rebooted, it had restored back to my last Timeshift, which was Dec 2020 when I updated my system.
4. I lost countless files, folders, etc.  My panel icons.  Everything was restored back to my immature system setup from Dec 2020.

**To Reproduce**
Steps to reproduce the behavior:  No idea.  I would imagine actually running Timeshift manually and restoring to that December 2020 backup.  Is there a log where I can examine what happened and figure out how to prevent it from happening again?

**Expected behavior**
I would not expect Timeshift to run without me manually doing it.  And if I did, I would expect some kind of confirmation before it restored everything that way.

**Screenshots**
None.  My last backup was mid-December 2020, and all of my files are restored back to that date.  Installed apps.  Keyboard settings.  I lost my known_hosts, and so much everywhere.

**System:**
 - Linux Mint 20 x86-64 5.4.0-26 kernel is where it happened, after I upgraded to 72
 - MATE
 - Ulyana

**Comment by RickCHodgin** [comment]
When I review my log, I see entries in there for Timeshift that I did not run, but were run on my system.  [http://www.libsf.org/misc/rick-hp-log.txt](rick-hp-log.txt)

If this was an attack against my machine, there should be some prompting that would enable me to verify that Timeshift would restore something before it happens.  And if there is none, then there needs to be a patch issued which adds it because this kind of attack should not be allowed to happen.  There should be some validation before Timeshift does something which restores data back.

Some kind of user approval should be required.

**Comment by teejee2008** [comment]
There's nothing in the log to indicate that a restore took place. The timeshift entry in the log file file is for a mount point /run/timeshift/backup that was unmounted by systemd.

Restoring a snapshot requires root access so it can't be executed by normal user programs.

**Comment by RickCHodgin** [comment]
You're not going to believe what it was.  There were two Linum Mint 20.0 partitions on my system, but only one was showing up via GRUB's boot menu.  When my normal / daily use 20.0 upgraded from Linux kernel ...26 to ...72, it rebuilt the GRUB boot menu and referenced the other 20.0 partition, and did not show my normal / daily use one.  I had created them both at the same time, and it was the same time as the Timeshift.

I went to reset my system thinking it was my 20.0 reset via Timeshift to that Dec 2020 version, and re-installing new software, etc.  I decided to upgrade to 20.1 since so much was lost anyway, and when I did, now a 20.1 showed up on the GRUB boot menu, and my normal / daily use 20.0 reappeared.

An unbelievable sequence of events yielding me to think Timeshift had reset everything.

This item can be closed.
